### ISSUE TITLE
Support oldrel/1, oldrel/2 etc.

### ISSUE THREAD

**Issue by gaborcsardi** [r-hub/rversions#28]
Like the API at https://api.r-hub.io/rversions#/paths/~1r-release/get

**Comment by gaborcsardi** [comment]
Maybe we should just switch to that API, because it has a cache.

**Comment by IndrajeetPatil** [comment]
This will be a great addition indeed!
I keep finding myself computing these old versions, especially in the context of check-full GHA.

What do you think about using a function factory in the package to create different versions of the underlying function?

``` r
library(rversions)

# internal function factory
.r_oldrel_n <- function(n) {
  function(dots = TRUE) {
    versions <- r_versions(dots)
    version_strs <- package_version(versions$version)
    major <- version_strs$major
    minor <- version_strs$minor
    major_minor <- unique(paste(major, sep = ifelse(dots, ".", "-"), minor))

    latest <- tail(major_minor, n)[1]

    tail(versions[versions$version < latest, ], 1)
  }
}

# exported functions
r_oldrel <- .r_oldrel_n(1)
r_oldrel_2 <- .r_oldrel_n(2)
r_oldrel_3 <- .r_oldrel_n(3)
r_oldrel_4 <- .r_oldrel_n(4)

# example usage (with dots)
r_oldrel() 
#>     version                date    nickname
#> 126   4.1.3 2022-03-10 08:05:38 One Push-Up

r_oldrel_2() 
#>     version                date        nickname
#> 122   4.0.5 2021-03-31 07:05:15 Shake and Throw

r_oldrel_3()
#>     version                date             nickname
#> 116   3.6.3 2020-02-29 08:05:16 Holding the Windsock

r_oldrel_4()
#>     version                date    nickname
#> 112   3.5.3 2019-03-11 08:04:49 Great Truth

# example usage (without dots)
r_oldrel(dots = FALSE) 
#>     version                date    nickname
#> 126   4-1-3 2022-03-10 08:05:38 One Push-Up

r_oldrel_2(dots = FALSE) 
#>     version                date        nickname
#> 122   4-0-5 2021-03-31 07:05:15 Shake and Throw

r_oldrel_3(dots = FALSE)
#>     version                date             nickname
#> 116   3-6-3 2020-02-29 08:05:16 Holding the Windsock

r_oldrel_4(dots = FALSE)
#>     version                date    nickname
#> 112   3-5-3 2019-03-11 08:04:49 Great Truth
```

<sup>Created on 2022-05-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

If you like this approach, happy to make a PR.

**Comment by IndrajeetPatil** [comment]
Created a PR: https://github.com/r-hub/rversions/pull/29

Happy to close it if you think this is not the way to go here.

**Comment by gaborcsardi** [comment]
Will be solved by #31 eventually.
